### PR TITLE
apiserver: switch charms to NewFacade/facade.Context

### PR DESCRIPTION
**Pull request by 4a6f656c**
Also remove pointless stateShim, unused Charms interface and unused
getState function. Rename charmsAccess to backend for consistency.

No functional change.

### CODE REVIEW

File: apiserver/charms/client.go [hunk 1]
```diff
@@ -16,17 +17,19 @@ import (
...
-}
-
 // Charms defines the methods on the charms API end point.
 type Charms interface {
 	List(args params.CharmsList) (params.CharmsListResult, error)
 	CharmInfo(args params.CharmURL) (params.CharmInfo, error)
 	IsMetered(args params.CharmURL) (bool, error)
 }
 
+type charmsAccess interface {
```
Comment (mjs) [inline apiserver/charms/client.go]: Could you renaming this "backend" while you're at it? That's more consistent with other (newer) facades.

File: apiserver/charms/client.go [hunk 2]
```diff
@@ -16,17 +17,19 @@ import (
 	"github.com/juju/juju/state"
 )
 
-var getState = func(st *state.State) charmsAccess {
-	return stateShim{st}
-}
-
 // Charms defines the methods on the charms API end point.
 type Charms interface {
```
Comment (mjs) [inline apiserver/charms/client.go]: I suspect this is unused and can be removed.

Comment (4a6f656c) [inline apiserver/charms/client.go]: Done.

File: apiserver/charms/client.go (hunk 1 above)
Comment (4a6f656c) [inline apiserver/charms/client.go]: Also done.

**Comment by 4a6f656c** [comment]
PTAL

**Comment by mjs** [comment]
$$merge$$

**Comment by mjs** [comment]
$$mongodb-died$$

**Pull request merged**